src/application.js

```diff
@@ -56,7 +56,7 @@ class Application {
             // Otherwise, update environment with new version
             if (env) {
                 winston.info(`Deploying ${versionLabel} to ${environmentName}...`);
-                yield this.environment.deploy(versionLabel, environmentName, config);
+                yield this.environment.deploy(versionLabel, environmentName, stack, config);
```

Comment:
I wonder what the use case is for this change.
If you want to use a new stack, you can probably create a new environment.

Reply:
Normally on production, we should do blue-green deployment so we have to create a new environment anyway.
On non-production environments, we can delete the old environment and create an environment with the new stack.

Reply:
this change is mainly for non-production environments. When we do a deployment, we override the current environment with the new version for simplicity. However, when we make a change to the solution stack name, it doesn't update the environment with the new solution stack.